infra/feast-operator/internal/controller/services/services.go

```diff
@@ -325,6 +325,13 @@ func (feast *FeastServices) createPVC(pvcCreate *feastdevv1alpha1.PvcCreate, fea
 }
 
 func (feast *FeastServices) setDeployment(deploy *appsv1.Deployment) error {
+	var volumes []corev1.Volume
+	if feast.Handler.FeatureStore.Spec.Services != nil {
+		volumes = feast.Handler.FeatureStore.Spec.Services.Volumes
+	}
+	if volumes == nil {
+		volumes = []corev1.Volume{} // Ensure it's an empty slice instead of nil
+	}
```

Comment:
this should not be added here. instead, consider a new function that handles volume additions. however, they should only be added to the Deployment object if `Spec.Services.Volumes` is not empty.